### PR TITLE
Attemp to fix the rule

### PR DESCRIPTION
**Pull request by TomasVotruba**
Hi @dingo-d , thanks for access. This is an attemp to fix the rule.
I've added the CI so we can see result right in the repository :+1:

### CODE REVIEW

File: utils/rector/src/Rector/Renaming/RenameClassNameRector.php [hunk 1]
```diff
@@ -71,16 +71,18 @@ public function getNodeTypes(): array
...
+        $oldName = $node->name->toLowerString();
+        if (! str_contains($oldName, '_')) {
+            return null;
+        }
 
-			// Take name parts, capitalize the first letter and return a string.
-			$newName = implode('', array_map(fn($namePart) => ucfirst($namePart), $names));
+        $names = explode('_', $oldName);
 
-			return $this->classRenamer->renameNode($node, [$oldName => $newName]);
```
Comment (TomasVotruba) [inline utils/rector/src/Rector/Renaming/RenameClassNameRector.php:81]: In this case, `classRename` is too heavy service for class renaming. It also depends on few internals that are related to original `RenameClassRector` rule.

File: utils/rector/src/Rector/Renaming/RenameClassNameRector.php [hunk 2]
```diff
@@ -71,16 +71,18 @@ public function getNodeTypes(): array
...
-		}
+        // Take name parts, capitalize the first letter and return a string.
+        $newName = implode('', array_map(fn($namePart) => ucfirst($namePart), $names));
 
-		return null;
+        $node->name = new Node\Identifier($newName);
```
Comment (TomasVotruba) [inline utils/rector/src/Rector/Renaming/RenameClassNameRector.php:84]: Simple node name assign does the trick :+1:

**Review by dingo-d — APPROVED** [review]

**Comment by dingo-d** [comment]
Awesome! Thank you so much for helping out. I'm still green with PHP parser so it takes me a while to figure out how to do things properly. The https://github.com/rectorphp/php-parser-nodes-docs/ helps tons! 🙂

**Comment by TomasVotruba** [comment]
Of course, take your time :+1: 

It took me personally around 6 months to have a little clue what is going on in AST :laughing: 

I think you're doing very well and understand the concepts correctly :clap: 

If you haven't got the book yet, there is a chapter about tokens and AST that gives great intoduction to this topic:
https://leanpub.com/rector-the-power-of-automated-refactoring

**Comment by dingo-d** [comment]
Thanks, I've purchased the book, I'm sure it will help me out a ton 🙂

**Pull request merged**